subprojects/core/src/main/java/org/gradle/process/internal/JvmOptions.java

```diff
@@ -49,6 +51,8 @@
     public static final String JMX_REMOTE_KEY = "com.sun.management.jmxremote";
     public static final String JAVA_IO_TMPDIR_KEY = "java.io.tmpdir";
 
+    private static final Pattern DEBUG_JVM_ARG = Pattern.compile("-agentlib:jdwp=transport=dt_socket,server=n,suspend=n,address=localhost:(\\d+)");
```

Comment:
❌ There's no guarantee that people will put the arguments in exactly that order or even supply all of them. The regex should be much more lenient and only parse the things it really needs, independent of order.

Reply:
I see. I've refactored the code to make it much more flexible as it used to be.